holoviews/tests/ui/bokeh/test_callback.py

```diff
@@ -495,36 +530,218 @@ def popup_form(name):
     expect(locator).to_be_visible()
 
 
+@skip_popup
+@pytest.mark.parametrize("tool", ["box_select", "lasso_select", "tap"])
+@pytest.mark.parametrize("popup_position", [
+    "top_right", "top_left", "bottom_left", "bottom_right",
+    "right", "left", "top", "bottom"
+])
+@pytest.mark.usefixtures("bokeh_backend")
+def test_stream_popup_position_selection1d(serve_hv, points, tool, popup_position):
+    def popup_form(index):
+        if index:
+            return f"# selection\n{len(index)} {popup_position}"
+
+    hv.streams.Selection1D(source=points, popup=popup_form, popup_position=popup_position)
+    points.opts(tools=[tool], active_tools=[tool])
+
+    page = serve_hv(points)
+    hv_plot = page.locator('.bk-events')
+    expect(hv_plot).to_have_count(1)
+
+    box = hv_plot.bounding_box()
+    if tool == "box_select":
+        # try to get it centered as possible
+        start_x, start_y = box['x'] + 90, box['y'] + 90
+        end_x, end_y = box['x'] + 170, box['y'] + 125
+        page.mouse.move(start_x, start_y)
+        hv_plot.click()
+        page.mouse.down()
+        page.mouse.move(end_x, end_y)
+        page.mouse.up()
+    elif tool == "lasso_select":
+        start_x, start_y = box['x'] + 1, box['y'] + box['height'] - 1
+        mid_x, mid_y = box['x'] + 1, box['y'] + 1
+        end_x, end_y = box['x'] + box['width'] - 1, box['y'] + 1
+        page.mouse.move(start_x, start_y)
+        hv_plot.click()
+        page.mouse.down()
+        page.mouse.move(mid_x, mid_y)
+        page.mouse.move(end_x, end_y)
+        page.mouse.up()
+    elif tool == "tap":
+        mid_x, mid_y = box['x'] + 1, box['y'] + 1
+        page.mouse.move(mid_x, mid_y)
+        hv_plot.click()
+
+    # Wait for popup to show
+    wait_until(lambda: expect(page.locator("#selection")).to_have_count(1), page)
+    locator = page.locator("#selection")
+    expect(locator).to_have_count(1)
+    expect(locator).not_to_have_text("selection\n0")
+
+    popup_box = locator.bounding_box()
+
+    distance_to_left = abs(popup_box['x'] - box['x'])
```

Comment:
Also, an helper function here could be nice:

```suggestion
    l, r, t, b = _get_popup_distances(popup_box, box)
```